readers/api/transport.go

```diff
@@ -39,16 +39,24 @@ const (
 	defLimit       = 10
 	defOffset      = 0
 	defFormat      = "messages"
+	thingToken     = "Thing "
+	userToken      = "Bearer "
 )
 
 var (
-	errUnauthorizedAccess = errors.New("missing or invalid credentials provided")
-	auth                  mainflux.ThingsServiceClient
+	errUnauthorizedAccess  = errors.New("missing or invalid credentials provided")
+	errCannotAuthorizeUser = errors.New("authorization failed")
+	errThingAccess         = errors.New("thing has no permission")
+	errUserAccess          = errors.New("user has no permission")
+	errWrongToken          = errors.New("incorrect or missing Authorization header")
+	thingsAuth             mainflux.ThingsServiceClient
+	usersAuth              mainflux.AuthServiceClient
 )
 
 // MakeHandler returns a HTTP handler for API endpoints.
-func MakeHandler(svc readers.MessageRepository, tc mainflux.ThingsServiceClient, svcName string) http.Handler {
-	auth = tc
+func MakeHandler(svc readers.MessageRepository, tc mainflux.ThingsServiceClient, ac mainflux.AuthServiceClient, svcName string) http.Handler {
+	thingsAuth = tc
+	usersAuth = ac
```

Comment:
Aren't users authenticated via Users service? Why using `AuthServiceClient` then, which b the name should connect to Auth service (which can't identify the user, just read the policies)?

Reply:
we are using `AuthServiceClient`  to check the token passed as `Authorization` header in HTTP request, token that was issued by `auth` when user was authenticated